src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,178 @@
+package duke.task;
+
+import duke.parser.MissingInputFieldException;
+
+/**
+ * Represents a task.
+ */
+public abstract class Task {
+    private boolean isDone;
+    protected TaskType type;
+    public static String delimiter;
```

Comment:
I like how you try to abide by the rule of using nouns for things and verbs for actions.